### PR TITLE
Implemented Dir.children

### LINKED ISSUE #4807: Dir.entries() returns both "." and ".." in its results
Calling Dir.entries() returns both "." and ".." in the result set, and there is no sort order defined.

This essentially makes one have to call:

```
result= Dir.entries( "/some/dir")
result.delete "."
result.delete ".."
```
Before the results become usable.

Is this intended behavior and/or there's room for improvement?

### PR DESCRIPTION
**Pull request by Sija**
Closes #4807

### CODE REVIEW

File: spec/std/dir_spec.cr [hunk 1]
```diff
@@ -257,6 +257,14 @@ describe "Dir" do
...
     filenames = Dir.entries(__DIR__)
     filenames.includes?("dir_spec.cr").should be_true
+    filenames.includes?(".").should be_true
+    filenames.includes?("..").should be_true
+  end
+
+  it "lists children" do
+    filenames = Dir.children(__DIR__)
+    filenames.includes?(".").should be_false
+    filenames.includes?("..").should be_false
```
Comment (jhass) [inline spec/std/dir_spec.cr]: How about comparisons comparing the output to `Dir.entries` here? Like checking that the size is the same minus two and that all returned items are also in the other.

**Comment by RX14** [comment]
How about a yielding version?

File: spec/std/dir_spec.cr (hunk 1 above)
Comment (akzhan) [inline spec/std/dir_spec.cr]: Example with `size - 2` knows about implementation details, and should be not used.

**Comment by Sija** [comment]
@RX14 I wouldn't mind but there's no yielding version of `Dir.entries` so why here?

File: spec/std/dir_spec.cr (hunk 1 above)
Comment (Sija) [inline spec/std/dir_spec.cr]: @jhass done.

**Comment by ysbaddaden** [comment]
It would be nice to interate without allocating HEAP memory. I guess `#entries` should allow the same.

**Comment by Sija** [comment]
@ysbaddaden I concur, yet it sounds like a topic for a separate PR.

**Comment by RX14** [comment]
@Sija Isn't the yielding version of `Dir.entries` `Dir.foreach`? I think thats's pretty confusing. But yes, we can leave clearing up the naming to a different PR.

**Comment by Sija** [comment]
@RX14 IIRC it comes from Ruby, but yeah, I'd say it is a bit confusing. Ruby has `Dir.each_child` method to complement `Dir.foreach` btw.

**Comment by RX14** [comment]
I think it's confusing having different names. We should have `entries` and `each_entry` and `children` and `each_child`.

**Comment by bew** [comment]
@RX14 and what should be `Dir.each`? should we remove it?

**Comment by RX14** [comment]
What's currently `Dir.foreach` would become `Dir.each_entry`.

**Comment by Sija** [comment]
@RX14 Why not two versions of `Dir.entries` and `Dir.children` with yielding and non-yielding overloads?

**Comment by RX14** [comment]
@Sija Because that's not how we do naming of yielding/collection members. Think of `String#chars` and `String#each_char`. There are plenty of other examples.

The reason for the existing convention is that the name is more descriptive: `foo.each_char do |char|` vs `foo.chars do |char|`.

**Comment by Sija** [comment]
@RX14 Fair point, it is more descriptive indeed.

**Pull request merged**